src/services/goals.js

```diff
@@ -1553,95 +1560,70 @@ export async function createOrUpdateGoalsForActivityReport(goals, reportId) {
-  // return sequelize.transaction(async (transaction) => {
-  //   try {
-  //     const reportsWithGoal = await ActivityReport.findAll({
-  //       attributes: ['id'],
-  //       include: [
-  //         {
-  //           attributes: ['id'],
-  //           model: Objective,
-  //           required: true,
-  //           as: 'objectivesWithGoals',
-  //           include: [
-  //             {
-  //               attributes: ['id'],
-  //               model: Goal,
-  //               required: true,
-  //               where: {
-  //                 id: goalId,
-  //               },
-  //               as: 'goal',
-  //             },
-  //           ],
-  //         },
-  //       ],
-  //       transaction,
-  //       raw: true,
-  //     });
-
-  //     const isOnReport = reportsWithGoal.length;
-  //     if (isOnReport) {
-  //       throw new Error('Goal is on an activity report and can\'t be deleted');
-  //     }
-
-  //     const objectiveTopicsDestroyed = await ObjectiveTopic.destroy({
-  //       where: {
-  //         objectiveId: {
-  //           [Op.in]: sequelize.literal(
-  //             `(SELECT "id" FROM "Objectives" WHERE "goalId" = ${sequelize.escape(goalId)})`,
-  //           ),
-  //         },
-  //       },
-  //       transaction,
-  //     });
-
-  //     const objectiveResourcesDestroyed = await ObjectiveResource.destroy({
-  //       where: {
-  //         objectiveId: {
-  //           [Op.in]: sequelize.literal(
-  //             `(SELECT "id" FROM "Objectives" WHERE "goalId" = ${sequelize.escape(goalId)})`,
-  //           ),
-  //         },
-  //       },
-  //       transaction,
-  //     });
-
-  //     const objectivesDestroyed = await Objective.destroy({
-  //       where: {
-  //         goalId,
-  //       },
-  //       transaction,
-  //     });
-
-  //     const grantGoalsDestroyed = await GrantGoal.destroy({
-  //       where: {
-  //         goalId,
-  //       },
-  //       transaction,
-  //     });
-
-  //     const goalsDestroyed = await Goal.destroy({
-  //       where: {
-  //         id: goalId,
-  //       },
-  //       transaction,
-  //     });
-
-  //     return {
-  //       goalsDestroyed,
-  //       grantGoalsDestroyed,
-  //       objectiveResourcesDestroyed,
-  //       objectiveTopicsDestroyed,
-  //       objectivesDestroyed,
-  //     };
-  //   } catch (error) {
-  //     auditLogger.error(
-  //  `${logContext.namespace} - Sequelize error - unable to delete from db - ${error}`
-  //  );
-  //     return 0;
-  //   }
-  // });
+export async function destroyGoal(goalIds) {
+  try {
+    const reportsWithGoal = await ActivityReport.findAll({
+      attributes: ['id'],
+      include: [
+        {
+          attributes: ['id'],
+          model: Goal,
+          required: true,
+          where: {
+            id: goalIds,
+          },
+          as: 'goals',
+        },
+      ],
+    });
+
+    const isOnReport = reportsWithGoal.length;
+    if (isOnReport) {
```

Comment:
This would be all or nothing?

Reply:
I'm not saying this is wrong just food for thought.

Reply:
Yeah I think that makes sense, it is only triggered on the button press in the RTR, and I don't think we want to handle only deleting one yet since we treat those goals as one goal everywhere in the UI. Maybe if it turns out to be a problem in the real world.